addons/website/static/tests/tours/website_style_edition.js

```diff
@@ -29,12 +45,54 @@ wTourUtils.registerWebsitePreviewTour("website_style_edition", {
     // a #t=... at the end then removes the old one.
     extra_trigger: 'iframe html:not(:has(link[href$="web.assets_frontend.min.css"]))',
     run: checkFontSize,
+}, {
+    content: "Open the color combinations area",
+    trigger: '.o_we_theme_presets_collapse we-toggler',
+}, {
+    content: "Open a color combination",
+    trigger: '.o_we_cc_preview_wrapper',
+}, {
+    content: "Edit the background color of that color combination",
+    trigger: '.o_we_theme_presets_collapse we-collapse .o_we_so_color_palette:eq(0)',
+}, {
+    content: "Choose a color",
+    trigger: `.o_we_color_btn[style*="background-color:${TARGET_BODY_BG_COLOR}"]`,
+}, {
+    content: "Check the body background color was properly adapted",
+    trigger: 'iframe body',
+    extra_trigger: `
+        .o_we_theme_presets_collapse we-collapse .o_we_so_color_palette:eq(0) .o_we_color_preview[style*="${TARGET_BODY_BG_COLOR}"],
```

Comment:
Just to know, is this `extra_trigger` the part of the code which is actually making it so it waits for the refresh before going further? It's only "true" after the reload, not when the user clicks on the color, right?

Reply:
Yes, indeed. The way our editor works at the moment is that widgets visual is only updated once the change has been applied. I did not like relying on the color that is applied as inline style because its format may depend on the browser (hence why I added a two-parts selector).

That behavior may change with OWL though, maybe we'll display the selected color on the widget, then reload the CSS, then re-update the widget color if necessary (as that would keep the robustness of the current system but be more reactive while as efficient thanks to OWL (not forcing a re-render for nothing)). We'll see: in that case, this test will fail on this step. We'll unify the way we wait for CSS reload at that time.